core/pta/veraison_attestation/cbor.c

```diff
@@ -0,0 +1,208 @@
+			size_t psa_nonce_len, const uint8_t *measurement_value,
+			size_t mv_len, UsefulBuf cbor_evidence_buffer)
+{
+	/* prepare usefulbufs because qcbor only accepts them */
+	UsefulBufC ubc_eat_profile = UsefulBuf_FromSZ(eat_profile);
+	UsefulBufC ubc_psa_implementation_id = { psa_implementation_id,
+						 psa_implementation_id_len };
+	UsefulBufC ubc_measurement_type = UsefulBuf_FromSZ(measurement_type);
+	UsefulBufC ubc_signer_id = { signer_id, signer_id_len };
+	UsefulBufC ubc_psa_instance_id = { psa_instance_id,
+					   psa_instance_id_len };
+	UsefulBufC ubc_psa_nonce = { psa_nonce, psa_nonce_len };
+	UsefulBufC ubc_measurement_value = { measurement_value, mv_len };
+	QCBOREncodeContext encode_ctx;
+	UsefulBufC cbor_evidence = { NULL, 0 };
+	QCBORError err;
+
+	QCBOREncode_Init(&encode_ctx, cbor_evidence_buffer);
+
+	QCBOREncode_OpenMap(&encode_ctx);
+
+	/* Profile Definition */
+	QCBOREncode_AddTextToMapN(&encode_ctx, PSA_PROFILE_DEFINITION,
+				  ubc_eat_profile);
+
+	/* Client ID */
+	QCBOREncode_AddInt64ToMapN(&encode_ctx, PSA_CLIENT_ID, psa_client_id);
+
+	/* Security Lifecycle */
+	QCBOREncode_AddInt64ToMapN(&encode_ctx, PSA_SECURITY_LIFECYCLE,
+				   psa_security_lifecycle);
+
+	/* Implementation ID */
+	QCBOREncode_AddBytesToMapN(&encode_ctx, PSA_IMPLEMENTATION_ID,
+				   ubc_psa_implementation_id);
+
+	/* Software Components */
+	QCBOREncode_OpenArrayInMapN(&encode_ctx, PSA_SW_COMPONENTS); /* [ */
+	QCBOREncode_OpenMap(&encode_ctx); /* { */
+	QCBOREncode_AddTextToMapN(&encode_ctx,
+				  PSA_SW_COMPONENT_MEASUREMENT_TYPE,
+				  ubc_measurement_type);
+	QCBOREncode_AddBytesToMapN(&encode_ctx,
+				   PSA_SW_COMPONENT_MEASUREMENT_VALUE,
+				   ubc_measurement_value);
+	QCBOREncode_AddBytesToMapN(&encode_ctx, PSA_SW_COMPONENT_SIGNER_ID,
+				   ubc_signer_id);
+	QCBOREncode_CloseMap(&encode_ctx); /* } */
+	QCBOREncode_CloseArray(&encode_ctx); /* ] */
+
+	/* Nonce */
+	QCBOREncode_AddBytesToMapN(&encode_ctx, PSA_NONCE, ubc_psa_nonce);
+
+	/* Instance ID */
+	QCBOREncode_AddBytesToMapN(&encode_ctx, PSA_INSTANCE_ID,
+				   ubc_psa_instance_id);
+
+	QCBOREncode_CloseMap(&encode_ctx);
+
+	err = QCBOREncode_Finish(&encode_ctx, &cbor_evidence);
+	if (err != QCBOR_SUCCESS) {
+		return NULLUsefulBufC;
+	} else {
+		return cbor_evidence;
+	}
+}
+
+UsefulBufC encode_protected_parameter(UsefulBuf buffer_for_protected_parameter);
+UsefulBufC create_tbs(UsefulBufC protected_parameters, UsefulBufC aad,
+		      UsefulBufC payload, UsefulBuf buffer_for_tbs);
+
+UsefulBufC generate_cose(UsefulBufC ubc_cbor_evidence,
+			 UsefulBuf buffer_for_cose)
+{
+	QCBOREncodeContext cose_context;
+	uint8_t signature[64];
+	size_t signature_len = 64;
+	UsefulBufC signature_payload = { signature, signature_len };
+	UsefulBufC protected_parameter = { NULL, 0 };
+	UsefulBuf_MAKE_STACK_UB(buffer_for_protected_parameter, 256);
+	UsefulBufC tbs_payload = { NULL, 0 };
+	UsefulBuf_MAKE_STACK_UB(buffer_for_tbs, 1024);
```

Comment:
1024 + 256 bytes on the stack, that's quite a lot, known that caller also allocates more than 512 byte on the stack also. Adding the 1024 byte stack buffer use by `xMSG()` macros.
For Arm32 machines, thread stack is deault 2048 byte, on Arm64, it is 8192 byte. Should the 32bit tack size be increased? Or maybe these tempoary buffers could also be alloced from `mempool_default`.

Reply:
I agree. As the code was originally developed as a PoC, it allocates a large memory. In practice, a smaller size should be required, so I will modify the code.

Reply:
We have made the required buffer size determined at runtime.